internal/install/packs/packs_installer.go

```diff
@@ -0,0 +1,137 @@
+package packs
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+
+	"github.com/newrelic/newrelic-cli/internal/credentials"
+	"github.com/newrelic/newrelic-cli/internal/install/types"
+	"github.com/newrelic/newrelic-cli/internal/install/ux"
+	"github.com/newrelic/newrelic-client-go/newrelic"
+	"github.com/newrelic/newrelic-client-go/pkg/dashboards"
+	"github.com/newrelic/newrelic-client-go/pkg/entities"
+	log "github.com/sirupsen/logrus"
+)
+
+// Maybe both fetchPacks and installPacks should be one interface
+// vs separate interfaces?
+// nolint: golint
+type PacksInstaller interface {
```

Comment:
Separate interfaces is fine, you can have a single concrete implementation that conforms to both if you like.  There is also now an `interfaces.go` that allows defining the interfaces closer to the usage (ie `install` package).  This will allow you to get rid of the `Concrete` prefix on the concrete type.

Reply:
Cool, I knew a single implementation could fulfill both interfaces somewhere in the back of my mind, just didn't think to do that. I kept the implementations separate for now just because it felt cleaner separating the concerns, but did shuffle the interface around and renamed `ConcretePacksInstaller` to `ServicePacksInstaller`. Moving the interfaces to `interfaces.go` displaces those outside of the `packs` package, and that seemed odd (plus the compiler was yelling at me and I couldn't quite figure out how to get it to resolve short of getting rid of the `packs` directory altogether).

Reply:
Happy to pair on this if you like, maybe I'm just missing something.